org.scala-ide.sdt.core/src/org/scalaide/core/IScalaProject.scala

```diff
@@ -222,6 +224,13 @@ trait IScalaProject extends Publisher[IScalaProjectEvent] {
    *  @param workingCopyOwner A working copy owner who's copies are searched first.
    */
   def newSearchableEnvironment(workingCopyOwner: WorkingCopyOwner = DefaultWorkingCopyOwner.PRIMARY): SearchableEnvironment
+
+  /**
+   * Says if compilation should be done in scopes e.g. main classes should be compiled
+   * before test classes. Used when compilation error in some scope should not break
+   * the compilation of dependent projects.
+   */
+  def compileInScopes(decision: Boolean): Unit
```

Comment:
How is this working? I didn't find any place where you save this value to a preference. Will it persist between opening and closing a project, or a restart?

I'd prefer for this option to be part of the `builder manager` section of project properties, like the other ones (compile order, stop build on error, etc.). That also ensures persistence.

Another problem with this flag (which I'd rather avoid if possible) is that `sbteclipse` does not know about it. So the default should be correct (probably `true`), so that people who didn't upgrade sbteclipse (assuming we'll update it, of course, to generate this flag as well) will get the right behavior. Utimately, if the heuristics are good enough, this flag would never be used.